image/refs/interface.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package refs implements generic name-based reference access.
+package refs
+
+import (
+	"github.com/opencontainers/image-spec/image/structure"
+)
+
+// Engine represents a name-based reference storage engine.
+type Engine interface {
+
+	// Put adds a new reference to the store.
+	Put(name string, descriptor *structure.Descriptor) (err error)
+
+	// Get returns a reference from the store.
+	Get(name string) (descriptor *structure.Descriptor, err error)
```

Comment:
Go os pkg returns an error on a file not found and provides [os.IsNotExists(err)](https://golang.org/pkg/os/#IsNotExist) the check it.

It depends if the not found should be considered an error or not.

I usually like to return a nil entry if I want to handle the not found case without relying on an error.

But in this case refsEnigine.Delete will return an error on missing name, so probably an IsNotExists(err) functions or a ErrNotFound should be defined anyway. 

And how the same case should be implemented in the casengine.Get function?